### PR TITLE
Add permissions for CEE to collect must-gathers in MGO namespace

### PR DESCRIPTION
**Pull request by csheremeta**
See https://issues.redhat.com/browse/OSD-5941

### CODE REVIEW

File: deploy/osd-must-gather-operator/01-osd-devaccess.Role.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: osd-devaccess
+  namespace: openshift-must-gather-operator
+rules:
+- apiGroups:
+  - ""
```
Comment (RiRa12621) [inline deploy/osd-must-gather-operator/01-osd-devaccess.Role.yaml:8]: Do they really need to be able to create **all** kinds of secrets?

Comment (csheremeta) [reply deploy/osd-must-gather-operator/01-osd-devaccess.Role.yaml]: no, I suppose not :)

File: deploy/osd-must-gather-operator/01-osd-devaccess.Role.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: osd-devaccess
+  namespace: openshift-must-gather-operator
+rules:
+- apiGroups:
+  - v1
```
Comment (jewzaam) [inline deploy/osd-must-gather-operator/01-osd-devaccess.Role.yaml:8]:
```suggestion
  - ""
```

File: deploy/osd-must-gather-operator/02-osd-devaccess.RoleBinding.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: osd-devaccess
+  namespace: openshift-must-gather-operator
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: osd-devaccess
```
Comment (jewzaam) [inline deploy/osd-must-gather-operator/02-osd-devaccess.RoleBinding.yaml:9]:
```suggestion
  name: osd-devaccess
  namespace: openshift-must-gather-operator
```

**Review by jewzaam — CHANGES_REQUESTED** [review]

File: deploy/osd-must-gather-operator/01-osd-devaccess.Role.yaml (hunk 2 above)
Comment (csheremeta) [reply deploy/osd-must-gather-operator/01-osd-devaccess.Role.yaml:8]: haha Rick suggested I change this :) I'll change it back

Comment (jewzaam) [reply deploy/osd-must-gather-operator/01-osd-devaccess.Role.yaml]: "v1" is the apiGroup version, it doesn't get put into roles.  The value of `apiGroups` is an array of apiGroup names.

**Comment by jewzaam** [comment]
/approve

Will let someone else +1 the PR, perhaps @RiRa12621 can take another pass through.

**Review by jewzaam — APPROVED** [review]

**Pull request merged**